editor/src/com/talosvfx/talos/editor/addons/scene/apps/routines/nodes/EventNodeWidget.java

```diff
@@ -0,0 +1,200 @@
+        super.read(json, jsonValue);
+
+        propertyWrappers.clear();
+        JsonValue propertiesJson = jsonValue.get("properties");
+        JsonValue propertyWrappersJson = propertiesJson.get("customParams");
+        if (propertyWrappersJson != null) {
+            for (JsonValue propertyWrapperJson : propertyWrappersJson) {
+                String className = propertyWrapperJson.getString("className", "");
+                JsonValue property = propertyWrapperJson.get("property");
+                if (property != null) {
+                    try {
+                        Class clazz = ClassReflection.forName(className);
+                        PropertyWrapper propertyWrapper = (PropertyWrapper) ClassReflection.newInstance(clazz);
+                        propertyWrapper.read(json, property);
+                        propertyWrappers.add(propertyWrapper);
+                    } catch (ReflectionException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+
+        fillFields();
+    }
+
+    @Override
+    protected void writeProperties(Json json) {
+        super.writeProperties(json);
+        json.writeArrayStart("customParams");
+        for (PropertyWrapper<?> propertyWrapper : propertyWrappers) {
+            json.writeObjectStart();
+            json.writeValue("className", propertyWrapper.getClass().getName());
+            json.writeValue("property", propertyWrapper);
+            json.writeObjectEnd();
+        }
+        json.writeArrayEnd();
+    }
+
+    private PropertyWrapper<?> createNewPropertyWrapper(PropertyType propertyType) {
+        PropertyWrapper<?> propertyWrapper = createPropertyInstanceOfType(propertyType);
+        return propertyWrapper;
+    }
+
+    private PropertyWrapper<?> createPropertyInstanceOfType(PropertyType type) {
+        try {
+            PropertyWrapper<?> propertyWrapper = PropertyTypeWrapperMapper.getWrapperForPropertyType(type).getConstructor().newInstance();
+            return propertyWrapper;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private void giveEmptyNameTo(PropertyWrapper<?> wrapper) {
+        Array<String> names = new Array<>();
+        for (PropertyWrapper tmp : propertyWrappers) {
+            if (tmp != wrapper) {
+                names.add(tmp.propertyName);
+            }
+        }
+
+        String suggestion = "value";
+        int iterator = 0;
+        while (names.contains(suggestion + iterator, false)) {
+            iterator++;
+        }
+
+        wrapper.propertyName = suggestion + iterator;
+    }
+
+    private void fillFields() {
+        fieldTable.clearChildren();
+
+        for (int i = 0; i < propertyWrappers.size; i++) {
+            PropertyWrapper<?> propertyWrapper = propertyWrappers.get(i);
+
+            try {
+                PropertyType type = propertyWrapper.getType();
```

Comment:
Why Try catch here? Probably better to obtain widget in another method that handles the exceptions so we dont have this large indenting in entire block covered by a pokemon exception

Reply:
fixed